content/projects/gsoc/2023/project-ideas/gitlab-plugin-modernization.adoc

```diff
@@ -0,0 +1,68 @@
+---
+layout: gsocprojectidea
+title: "GitLab Plugin Modernization"
+goal: "Cleaning and modernizing the extensively used GitLab plugin"
+category: Plugin improvement
+year: 2023
+status: draft
+sig: platform
+skills:
+- Java
+- Docker
+- GitLab
+mentors:
+- "gsoc"
+
+links:
+    emailThread: https://community.jenkins.io/t/gsoc-2023-project-idea-active-modernization-of-gitlab-plugin/5039
+//   gitter: "jenkinsci/plugin-installation-manager-cli-tool"
+//   draft: https://docs.google.com/document/d/1s-dLUfU1OK-88bCj-GKaNuFfJQlQNLTWtacKkVMVmHc
+---
+=== Background
+
+The GitLab plugin is widely used (58,727 installations) but has not had an active maintainer for a long time. 
+Many of us at CloudBees have volunteered to keep it afloat over the years, including Owen, Basil, Bruno, Jean-Marc, and Mark. 
+But it could use a sprint of active development. 
+Maybe GSoC could be a good way to staff this with an intern?
+
+At first sight, there are a number of things that need to be done to keep this plugin afloat in the long term:
+
+* Migrate from the very old version of RESTEasy it is using to the GitLab API plugin (just like GitLab Authentication, GitLab Branch Source, and GitLab Logo)
+* Revive the Docker-based functional test suite. It has been broken for some time and is currently disabled.
+* Fix any of the dozens of outstanding bugs.
+
+The objectives could be expanded to cover the "sister" `gitlab-branch-source` plugin.
```

Comment:
In what way? I am not aware of any pending tasks for `gitlab-branch-source` that need delegation.

Reply:
reverted back to "discussion" via #5964